qutebrowser/completion/completionwidget.py

```diff
@@ -311,10 +314,15 @@ def set_model(self, model):
         for i in range(model.rowCount()):
             self.expand(model.index(i, 0))
 
-    def set_pattern(self, pattern):
+    def set_pattern(self, pattern: str) -> None:
         """Set the pattern on the underlying model."""
         if not self.model():
             return
+        if self.pattern == pattern:
+            # no changes, abort
+            log.completion.debug(
+                "Ignoring pattern set request as pattern has not changed.")
+            return
```

Comment:
I think skipping `_maybe_update_geometry` might be unsafe here, it seems like the size is re-used if you open, close, and open again. Maybe if I clear out the pattern when the completion window is closed, this would be safe.